torch/distributed/distributed_c10d.py

```diff
@@ -2443,6 +2466,12 @@ def new_group(ranks=None, timeout=default_pg_timeout, backend=None):
     # barrier at the end to ensure that once we return from this method, all
     # process groups including global variables are updated correctly on all
     # ranks.
-    barrier()
+    if backend == Backend.MPI:
```

Comment:
there might be other c10d extensions besides MPI. Does it make sense if we only do `_store_based_barrier` for NCCL backend?

Reply:
This follows the same logic in other parts of the code where MPI assumes there is no store and all other backends have to initialize a store.